docs/io/index_unofficial.rst

```diff
@@ -0,0 +1,287 @@
+
+In an attempt to simplify and standardise the process of creating a new
+reader, this documentation has been created, containing the guidelines to be
+followed. These guidelines can be summarised as:
+
+* Each reader should have a corresponding
+  `ctapipe.io.eventfilereader.EventFileReader` in
+  `ctapipe.io.unofficial.eventfilereader` to ensure the common high-level
+  interface for reading data is maintained. This is explained in
+  `EventFileReaders`_.
+* Care should be taken in the imports of external modules. There should be no
+  dependencies added to ctapipe on the external libraries required for these
+  unofficial readers. The correct approach to achieve this is
+  explained in `EventFileReaders`_.
+
+
+How to use my reader in ctapipe?
+================================
+
+Presumably you already have some software that exists to read in your data
+format. What form that reader exists in dictates how easy it is to incorporate
+into ctapipe. Some of the possible forms and appropriate inplementations are:
+
+* **Your reader can be entirely programmed in Python**, in which case your
+  reader does not depend on any external packages, and you can include that
+  code in your reader's `Low-Level Reader-Specific Modules`_.
+* **Your reader exists as an external Python module**, in which case it could
+  be made available on conda for easy (optional) installation.
+* **You have a C/C++ inplementation of your reader in a simple script**,
+  therefore you could provide an interface to that functionality by using
+  "ctypes". The C/C++ scripts can exist inside your reader module, and are
+  compiled as part of the ctapipe installation. See
+  `ctapipe.utils.neighbour_sum` as an example of how to interface Python with
+  C code, and remember to add your extension to ext_modules in setup.py.
+* **Your reader exists as an external C/C++ library**, therefore software such
+  as SWIG can be used to provide a Python wrapper to the library,
+  automatically creating interfaces to the functions in the library.
+
+
+EventFileReaders
+================
+
+The module `ctapipe.io.unofficial.eventfilereader` contains the
+`ctapipe.io.eventfilereader.EventFileReader` classes for the unofficial
+readers. These classes provides the high-level interface for reading data
+from each data format. Read about the purpose of EventFileReaders at
+:ref:`EventFileReaderClasses`.
+
+Each reader is required to import their
+`Low-Level Reader-Specific Modules`_ inside their `__init__` method,
+therefore bypassing the need for the external reader software to be installed
+unless someone has chosen to use this specific
+`ctapipe.io.eventfilereader.EventFileReader`, in which case a
+`ModuleNotFoundError` will be raised if the external software has not been
+found.
+
+The reader must also implement a fast method for
+`ctapipe.io.eventfilereader.EventFileReader.is_compatible`. This
+method attempts to find a reader which *may* be compatible with the input
```

Comment:
I think you made a mistake here. A method `EventFileReader.is_compatible` should only return a bool, wheiter a given path is compatible or not. It should not find a compatible reader. It  may be used somewhere else to find a compatible reader.
E.g. like this:
```python
for reader in readers:
    try:
        if reader.is_compatible(path):
            break
    except ModuleNotFoundError:
        pass
else:
    raise IOError("No compatible reader found for path")
```